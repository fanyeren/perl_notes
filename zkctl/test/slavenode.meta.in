#!/bin/bash
source $(dirname $0)/common

NODEID="hjz"
node_clear slavenode
./zkctl -w slavenode add --slavenode-id=$NODEID --data={}

raw='"cpu_capacity":"1000","disk_capacity":"1200G","disk_path":"/home","failure_domain":"default","flash_capacity":"640G","flash_path":"/ssd1","group":"group_1","idc":"tc","online_state":"online","safe-mode":"off","work_path":"/home/work/search"'

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w slavenode add-meta-attr --slavenode-id=$NODEID --key=${kv[0]} --value=${kv[1]}
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -s slavenode get-meta-attr --slavenode-id=$NODEID --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -s slavenode get-meta-all --slavenode-id=$NODEID
echo "got all cluster result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w slavenode del-meta-attr --slavenode-id=$NODEID --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -s slavenode get-meta-all --slavenode-id=$NODEID
echo "got empty cluster result result=$?"
