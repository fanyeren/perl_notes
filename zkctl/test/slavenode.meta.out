Add meta cpu_capacity=1000 result 0
1000
Get meta cpu_capacity=1000 result 0
Add meta disk_capacity=1200G result 0
1200G
Get meta disk_capacity=1200G result 0
Add meta disk_path=/home result 0
/home
Get meta disk_path=/home result 0
Add meta failure_domain=default result 0
default
Get meta failure_domain=default result 0
Add meta flash_capacity=640G result 0
640G
Get meta flash_capacity=640G result 0
Add meta flash_path=/ssd1 result 0
/ssd1
Get meta flash_path=/ssd1 result 0
Add meta group=group_1 result 0
group_1
Get meta group=group_1 result 0
Add meta idc=tc result 0
tc
Get meta idc=tc result 0
Add meta online_state=online result 0
online
Get meta online_state=online result 0
Add meta safe-mode=off result 0
off
Get meta safe-mode=off result 0
Add meta work_path=/home/work/search result 0
/home/work/search
Get meta work_path=/home/work/search result 0
{"cpu_capacity":"1000","disk_capacity":"1200G","disk_path":"\/home","failure_domain":"default","flash_capacity":"640G","flash_path":"\/ssd1","group":"group_1","idc":"tc","online_state":"online","safe-mode":"off","work_path":"\/home\/work\/search"}
got all cluster result result=0
del meta attr cpu_capacity result=0
del meta attr disk_capacity result=0
del meta attr disk_path result=0
del meta attr failure_domain result=0
del meta attr flash_capacity result=0
del meta attr flash_path result=0
del meta attr group result=0
del meta attr idc result=0
del meta attr online_state result=0
del meta attr safe-mode result=0
del meta attr work_path result=0
{}
got empty cluster result result=0
